draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -1198,6 +1198,7 @@ https://www.example.org
    by network failure, a server &SHOULD; generate encoding or
    length-delimited messages whenever possible.  The close-delimiting
    feature exists primarily for backwards compatibility with HTTP/1.0.
+   Note that a half-closed TCP connection does not delimit a request.
```

Comment:
This note appears out of nowhere.

I think it would be better to change the beginning sentence to add "response" above, as in

```
   Since there is no way to distinguish a complete close-delimited response
   message from a partially received response interrupted
   by network failure, a server &SHOULD; generate encoding or
   length-delimited response messages whenever possible.
   This close-delimiting
   feature exists primarily for backwards compatibility with HTTP/1.0.

```
and then add the note as a paragraph:
```suggestion
   Note: Request messages are never close-delimited because they are always
   explicitly framed by length or transfer coding, with the absence of both implying
   the request ends immediately after the header section. A half-closed TCP
   connection does not delimit a request message, nor does it imply that the client
   is no longer interested in a response. In general, do not rely on transport signals
   to signal edge cases, since HTTP/1.1 is independent of transport.
```